jmespath-core/src/main/java/io/burt/jmespath/jcf/JcfAdapter.java

```diff
@@ -143,6 +154,15 @@ public Object getProperty(Object value, String name) {
   }
 
   @Override
+  public Collection<String> getPropertyNames(Object value) {
+    if (isObject(value)) {
+      return ((Map<String, Object>) value).keySet();
```

Comment:
It was meant as a (bad) argument for changing the `getPropertyNames`. Or I suppose rather a counter-counter-argument to an expected counter-argument that never came. I wanted to highlight that this implementation in fact implements an (in the general case illegal) shortcut, which, for example, a JRuby implementation could not do. Feel free to ignore my comment.